pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +194,74 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Compute command to execute spot-to-spot consolidation if:
+//  1. The SpotToSpotConsolidation feature flag is set to true.
+//  2. For single-node consolidation:
+//     a. There are at least 15 cheapest instance type replacement options to consolidate.
+//     b. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate, results *pscheduling.Results,
+	candidatePrice float64) (Command, error) {
+
+	// Spot consolidation is turned off.
+	if !options.FromContext(ctx).FeatureGates.SpotToSpotConsolidation {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
+		}
+		logging.FromContext(ctx).Debugf("SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node.")
+		return Command{}, nil
+	}
+
+	// All possible replacements for the current candidate filtered by spot offerings
+	instanceTypeOptionsWithSpotOfferings :=
+		results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions.FilterByCapacityType(results.NewNodeClaims[0].Requirements, v1beta1.CapacityTypeSpot)
+
+	// Allow single-node only if there are minimum number of possible InstanceTypes for a NodeClaim
+	if len(candidates) == 1 && len(instanceTypeOptionsWithSpotOfferings) < MinInstanceTypesForSpotToSpotConsolidation {
+		// Intentionally publishing only for single-node inorder to avoid triggering a bunch of events to
+		c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, fmt.Sprintf("SpotToSpotConsolidation requires %d instance type replacement options to consolidate",
+			MinInstanceTypesForSpotToSpotConsolidation))...)
```

Comment:
Thoughts about adding this? It has the downside of de-duping less often, but could give more information on how far off they are from instance type replacement options. It wouldn't be uncommon for successive disruption loops that this number could be the same, since it would be the same instance types. 
```suggestion
		c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, fmt.Sprintf("SpotToSpotConsolidation requires %d instance type replacement options to consolidate, got %d",
			MinInstanceTypesForSpotToSpotConsolidation), len(instanceTypeOptionsWithSpotOfferings)...)
```

Reply:
Yes, we can do this. I actually added this information in the logs to help the customers. Surely we can add this in the events too. Addressed this.